packages/workflow-engine-redis/src/services/workflow-orchestrator.ts

```diff
@@ -110,7 +117,16 @@ export class WorkflowOrchestratorService {
   }
 
   async onApplicationShutdown() {
+    this.logger.info("Workflow orchestrator gradeful shutdown.")
+    // set flag to shuting down
+    this.isShuttingDown = true
+    while (this.activeStepsCount > 0) {
+      this.logger.info(`Active steps: ${this.activeStepsCount}`)
+      // sleep for 1 second
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+    }
     await this.redisDistributedTransactionStorage_.onApplicationShutdown()
+    this.logger.info("Workflow orchestrator finished!")
```

Comment:
thought: Not sure we should log all this stuff, great for debugging, but not super useful for other users
```suggestion
    this.isShuttingDown = true
    while (this.activeStepsCount > 0) {
      await new Promise((resolve) => setTimeout(resolve, 1000))
    }
    await this.redisDistributedTransactionStorage_.onApplicationShutdown()
```